### ISSUE TITLE
Add `reactLibrary` setting

### ISSUE THREAD

**Issue by cherryblossom000** [yannickcr/eslint-plugin-react#2904]
Add a `reactLibrary` setting for the name of the React library (defaults to `'react'`). This would be helpful for [Preact](https://preactjs.com/) users.

Currently, the `jsx-fragments` rule (set to `syntax`) doesn't warn about this:

```jsx
import {Fragment} from 'preact'

const foo = <Fragment>foo</Fragment>
```

Checking for the configured `reactLibrary` setting instead of the hard-coded `'react'` would make this rule (and possibly others, I haven't really looked into this plugin that much) more flexible.

**Comment by ljharb** [comment]
This already exists: https://github.com/yannickcr/eslint-plugin-react#configuration see “pragma”

**Comment by cherryblossom000** [comment]
@ljharb That only works for detecting `<Preact.Fragment>` and not `Fragment` if it was imported from `'preact'`.

As you can see here, for imports the rule only checks if the imported module is `'react'`:

https://github.com/yannickcr/eslint-plugin-react/blob/0e9a193435454cd4891760d2bc5efdb2d2bc6294/lib/rules/jsx-fragments.js#L115-L124

https://github.com/yannickcr/eslint-plugin-react/blob/0e9a193435454cd4891760d2bc5efdb2d2bc6294/lib/rules/jsx-fragments.js#L155-L165

Here's a minimal example of my issue:

`.eslintrc.yml`
```yml
root: true
plugins: [react]
parserOptions:
  ecmaFeatures:
    jsx: true
  ecmaVersion: 6
  sourceType: module
settings:
  react:
    pragma: Preact
rules:
  react/jsx-fragments: 2
```

`test.js`
```js
import {Fragment} from 'preact'

const foo = <Fragment>foo</Fragment>
```

Running `npx eslint test.js` does not report any linting errors.

eslint version: `7.18.0`
eslint-plugin-react version: `7.22.0`

**Comment by ljharb** [comment]
In that case, the rule definitely shouldn’t be hardcoded - i understand what you mean now.

A PR to add this setting would be appreciated.
